core/src/main/scala/org/http4s/EntityEncoder.scala

```diff
@@ -147,11 +147,23 @@ trait EntityEncoderInstances extends EntityEncoderInstances0 {
   implicit val byteArrayEncoder: EntityEncoder[Array[Byte]] =
     chunkEncoder.contramap(Chunk.bytes)
 
-  // TODO fs2 port this is where we miss ByteVector
-  /*
   implicit val byteBufferEncoder: EntityEncoder[ByteBuffer] =
-    chunkEncoder.contramap(ByteVector.view)
-   */
+    chunkEncoder.contramap { bb =>
+      val duplicate = bb.duplicate()
+      // This is triggered if puts have been done, then we want to flip to read
+      if (duplicate.remaining != duplicate.capacity()) {
+        duplicate.flip()
```

Comment:
I don't think I like this: it's not an uncommon pattern with `ByteBuffer`s for someone to read bytes from a buffer (maybe a length field or something) and then want to use the rest elsewhere.